cmd/kubeadm/app/util/config/masterconfig.go

```diff
@@ -56,9 +56,17 @@ func SetInitDynamicDefaults(cfg *kubeadmapi.InitConfiguration) error {
 	// This is the same logic as the API Server uses
 	ip, err := netutil.ChooseBindAddress(addressIP)
 	if err != nil {
-		return err
+		switch err.(type) {
+		case *netutil.NoRoutesError:
+			defaultBind := "127.0.0.1"
+			glog.Warningf("could not obtain advertise address for the API Server: %v; using: %s", err, defaultBind)
+			cfg.API.AdvertiseAddress = defaultBind
```

Comment:
this will lead to the error. net space inside container with apiserver will be not the same as host, thus if we can't get bind ip, init must fail. What we can do, is to try to keep AdvertiseAddress empty, and in places where it is important, do the checks for it to have some normal values.

Reply:
@kad i was hesitant about adding 127.0.0.1.
i will give `""` a try.

Reply:
but please have a check that it is valid in the places where it is really used. e.g. with https://golang.org/pkg/net/#IP.IsGlobalUnicast